runtime/memprof.c

```diff
@@ -1880,16 +1881,16 @@ static void maybe_track_block(memprof_domain_t domain,
   set_action_pending_as_needed(domain);
 }
 
-/* Renew the next sample in a domain's minor heap. Could race with
- * sampling and profile-stopping code, so do not call from another
- * domain unless the world is stopped. Must be called after each minor
- * sample and after each minor collection. In practice, this is called
- * at each minor sample, at each minor collection, and when sampling
- * is suspended and unsuspended. Extra calls do not change the
- * statistical properties of the sampling because of the
- * memorylessness of the geometric distribution. */
+/* Sets the trigger for the next sample in a domain's minor
+ * heap. Could race with sampling and profile-stopping code, so do not
+ * call from another domain unless the world is stopped. Must be
+ * called after each minor sample and after each minor collection. In
+ * practice, this is called at each minor sample, at each minor
+ * collection, and when sampling is suspended and unsuspended. Extra
+ * calls do not change the statistical properties of the sampling
+ * because of the memorylessness of the geometric distribution. */
 
-void caml_memprof_renew_minor_sample(caml_domain_state *state)
+void caml_memprof_set_trigger(caml_domain_state *state)
```

Comment:
At present, no, this is only called from the same domain. But I wanted to make the constraints clear here in case that might change in the future. Possibly the constraint should also be documented in a comment in `memprof.h`.